main.go

```diff
@@ -137,46 +145,79 @@ func libhoneyConfig() libhoney.ClientConfig {
 		log.Warnln("LIBHONEY_API_KEY or LIBHONEY_DATASET not set, disabling libhoney")
 		return libhoney.ClientConfig{}
 	}
-	userAgent := fmt.Sprintf("honeycomb-lambda-extension-%s/%s", runtime.GOARCH, version)
 
 	return libhoney.ClientConfig{
-		APIKey:  apiKey,
-		Dataset: dataset,
-		APIHost: apiHost,
-		Transmission: &transmission.Honeycomb{
-			MaxBatchSize:          libhoney.DefaultMaxBatchSize,
-			BatchTimeout:          libhoney.DefaultBatchTimeout,
-			MaxConcurrentBatches:  libhoney.DefaultMaxConcurrentBatches,
-			PendingWorkCapacity:   libhoney.DefaultPendingWorkCapacity,
-			UserAgentAddition:     userAgent,
-			EnableMsgpackEncoding: true,
-		},
+		APIKey:       apiKey,
+		Dataset:      dataset,
+		APIHost:      apiHost,
+		Transmission: newTransmission(),
+	}
+}
+
+func newTransmission() *transmission.Honeycomb {
+	batchSendTimeout := envOrElseDurationSeconds("HONEYCOMB_BATCH_SEND_TIMEOUT_S", defaultBatchSendTimeout)
+
+	userAgent := fmt.Sprintf("honeycomb-lambda-extension-%s/%s", runtime.GOARCH, version)
+
+	return &transmission.Honeycomb{
+		MaxBatchSize:          libhoney.DefaultMaxBatchSize,
+		BatchTimeout:          libhoney.DefaultBatchTimeout,
+		MaxConcurrentBatches:  libhoney.DefaultMaxConcurrentBatches,
+		PendingWorkCapacity:   libhoney.DefaultPendingWorkCapacity,
+		UserAgentAddition:     userAgent,
+		EnableMsgpackEncoding: true,
+		BatchSendTimeout:      batchSendTimeout,
 	}
 }
 
-// helper functions for environment variables with default fallbacks
+// (helper) Retrieve an environment variable value by the given key,
+// return an integer based on that value.
+// If key cannot be found or value fails to cast to an int,
+// return the given fallback duration.
 func envOrElseInt(key string, fallback int) int {
 	if value, ok := os.LookupEnv(key); ok {
 		v, err := strconv.Atoi(value)
 		if err != nil {
+			log.Warnf("%s was provided, but failed to parse to an integer. Falling back to default of %d.", key, fallback)
```

Comment:
in these warnings, why not also print the value: "%s was provided as '%s', but this failed..."?

Reply:
I suspect my inherent distrust of redisplaying user input. I think your point is good, yes, this would be helpful feedback to the configuring administrator.

Reply:
Updated in https://github.com/honeycombio/honeycomb-lambda-extension/pull/98/commits/1ff3c98d2fe896607f9038b613d5bceb92bbc6e7